### PR TITLE
Added C++, Java and Python Code of Longest Common SubString. Issue #667

### PR DESCRIPTION
**Pull request by prashant-mahanta**
Added C++, Java and Python Code of Longest Common SubString.
Issue: [#667](https://github.com/jainaman224/Algo_Ds_Notes/issues/667)

### CODE REVIEW

File: Longest_Common_Substring/longest_common_substring.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
+// Space Complexity: O(n)
```
Comment (jainaman224) [inline Longest_Common_Substring/longest_common_substring.cpp]: Fix file name.

File: Longest_Common_Substring/longest_common_substring.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+	cout << "String1: " << string1 << "\nString2: " << string2 << "\n";
+	if(lcsStr == "-1"){
+		cout << "No common substring\n"; 
+	}
+	else{
+		cout << "Longest Common Substring: " << lcsStr << " (of length: " << lcsStr.size() << ")\n";
+	}
+	return 0;
+}
```
Comment (jainaman224) [inline Longest_Common_Substring/longest_common_substring.cpp]: Add sample output.

File: Longest_Common_Substring/longest_common_substring.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	consqRow = []
+	for i in range(2):
+		temp = []
+		for j in range(n+1):
+			temp.append(0)
+		consqRow.append(temp)
+	
+
+	# 1st row and 1st column matrix consqRow need to be 0
+    # As the default value initialized is 0 so we can ignore this condition here
```
Comment (jainaman224) [inline Longest_Common_Substring/longest_common_substring.py]: Fix indentation

**Comment by prashant-mahanta** [comment]
@jainaman224 I have fixed the issues. Please have a look at it.

File: Longest_Common_Substring/Longest_Common_Substring.cpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
+// Space Complexity: O(n)
+// Time Complexity: O(m*n) 
+
+#include<bits/stdc++.h>
+using namespace std;
+	
+
+string LongestCommonSubstring(string str1, string str2) 
+{ 
+
```
Comment (jainaman224) [inline Longest_Common_Substring/Longest_Common_Substring.cpp]: Unnecessary new line.

File: Longest_Common_Substring/Longest_Common_Substring.cpp [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+		str2 = temp;
+	}
+	int m, n; 
+	m = str1.size();
+	n = str2.size();
+
+	int maxlength = 0; //length of longest common Substring
+	int end; //ending point of longest common Substring
+
+
```
Comment (jainaman224) [inline Longest_Common_Substring/Longest_Common_Substring.cpp]: Unnecessary new line.

File: Longest_Common_Substring/Longest_Common_Substring.cpp [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+	vector< vector<int> > consqRow(2,vector<int> (n+1,0)); //store result of 2 consecutive rows
+	int curr = 0; //current row in the matrix  
+
+
```
Comment (jainaman224) [inline Longest_Common_Substring/Longest_Common_Substring.cpp]: Unnecessary new line.

File: Longest_Common_Substring/Longest_Common_Substring.cpp [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+			} 
+			else { 
+				consqRow[curr][j] = 0; 
+			} 
+		} 
+
+		curr = 1 - curr; // changing the row alternatively
+	} 
+  	
+
```
Comment (jainaman224) [inline Longest_Common_Substring/Longest_Common_Substring.cpp]: Too many new lines.

File: Longest_Common_Substring/Longest_Common_Substring.cpp [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+	cout << "String1: " << string1 << "\nString2: " << string2 << "\n";
+
+	if(lcsStr == "-1"){
+		cout << "No common substring\n"; 
+	}
+	else{
+		cout << "Longest Common Substring: " << lcsStr << " (of length: " << lcsStr.size() << ")\n";
+	}
+	return 0;
+
```
Comment (jainaman224) [inline Longest_Common_Substring/Longest_Common_Substring.cpp]: Unnecessary new line.

**Comment by prashant-mahanta** [comment]
@jainaman224 I have corrected it. Please have a look at it.

**Pull request merged**